fusor-ember-cli/app/controllers/review/progress/overview.js

```diff
@@ -52,6 +53,44 @@ export default Ember.Controller.extend(ProgressBarMixin, NeedsDeploymentMixin, {
     }
 
     return true;
-  })
+  }),
+
+  isAbandonModalOpen: false,
+  loadingRedeployment: false,
+
+  actions: {
+    redeploy() {
+      this.set('loadingRedeployment', true);
 
+      let depl = this.get('deploymentController.model');
+      let token = Ember.$('meta[name="csrf-token"]').attr('content');
+
+      request({
+        url: '/fusor/api/v21/deployments/' + depl.get('id') + '/redeploy',
+        type: "PUT",
+        headers: {
+          "Accept": "application/json",
+          "Content-Type": "application/json",
+          "X-CSRF-Token": token,
+          "Authorization": "Basic " + this.get('session.basicAuthToken')
+        }
+      }).then((response) => {
+        let newTaskUUID = response.id;
+        depl.set('foreman_task_uuid', newTaskUUID);
+        depl.set('has_content_error', false);
+        depl.save();
+        this.send('refreshModelOnOverviewRoute');
+      }).catch((err) => {
+        console.log('ERROR occurred attempting a redeploy', err);
+      }).finally(() => this.set('loadingRedeployment', false));
+    },
+    abandon() {
+      this.set('isAbandonModalOpen', true);
+    },
+    executeAbandonment() {
+      let depl = this.get('deploymentController.model');
+      depl.destroyRecord();
```

Comment:
If it's not clear @isratrade, the `Abandon | Redeploy` prompt is not a modal, it's in-interface and replaces the progress bars / hides the cancel/back/deploy component. A simple confirmation modal is presented if you choose abandon. I'll post a screenshot when I get into the office. Maybe I need to be more explicit with the copy to make sure you intend to delete the deployment?

I'm not sure what the purpose of cancel would be if presented alongside Abandon and Redeploy.